yugabyte_cloudformation.yaml

```diff
@@ -0,0 +1,490 @@
+    CreationPolicy:
+      ResourceSignal:
+        Count: '1'
+        Timeout: PT5M
+    Metadata:
+       'AWS::CloudFormation::Init':
+        configSets:
+          InstallAndRun:
+            - Install
+            - Configure
+        Install:
+          packages:
+            yum:
+              git: []
+        Configure:
+          files:
+            /home/ec2-user/install_software.sh:
+              source: "https://raw.githubusercontent.com/vp15591/yugabytedb-cloudformation/master/scripts/install_software.sh"
+              mode: '0755'
+              owner: ec2-user
+              group: ec2-user
+            /home/ec2-user/create_universe.sh:
+              source: "https://raw.githubusercontent.com/vp15591/yugabytedb-cloudformation/master/scripts/create_universe.sh"
+              mode: '0755'
+              owner: ec2-user
+              group: ec2-user    
+            /home/ec2-user/start_master.sh:
+              source: "https://raw.githubusercontent.com/vp15591/yugabytedb-cloudformation/master/scripts/start_master.sh"
+              mode: '0755'
+              owner: ec2-user
+              group: ec2-user
+            /home/ec2-user/start_tserver.sh:
+              source: "https://raw.githubusercontent.com/vp15591/yugabytedb-cloudformation/master/scripts/start_tserver.sh"
+              mode: '0755'
+              owner: ec2-user
+              group: ec2-user 
+          commands:
+            01_Install_YugaByte_DB:
+              command:  bash /home/ec2-user/install_software.sh !FindInMap [ YugaByteConfig, Edition, Value ] !Ref DBVersion !FindInMap [ YugaByteConfig, Download, URL ]
+            02_Create_universe:
+              command: !Sub |
+                bash /home/ec2-user/create_universe.sh AWS ${AWS::Region} 3 ${Node0NetInt.PrimaryPrivateIpAddress} ${Node1NetInt.PrimaryPrivateIpAddress} ${Node2NetInt.PrimaryPrivateIpAddress} ${PublicSubnet0.AvailabilityZone} ${PublicSubnet1.AvailabilityZone} ${PublicSubnet2.AvailabilityZone} ${PublicSubnet1.AvailabilityZone}
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: !Ref InstanceType
+      KeyName: !Ref KeyName
+      NetworkInterfaces : 
+        - NetworkInterfaceId: !Ref Node1NetInt
+          DeviceIndex: 0
+      Tags:
+        - Key: "Name"
+          Value: !Join
+            - ''
+            - - !Ref "AWS::StackName"
+              - 'Node-1'
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash -xe
+            yum update -y aws-cfn-bootstrap
+            /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource YugaByteNode1 --configsets InstallAndRun --region ${AWS::Region}
+            /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource YugaByteNode1 --region ${AWS::Region}
+      BlockDeviceMappings:
+        - DeviceName: "/dev/xvda"
+          Ebs:
+            VolumeSize: 50
+            DeleteOnTermination: "true"
+            VolumeType: "gp2"            
+
+
+  YugaByteNode2:
```

Comment:
why are we duplicating the YugaByteNodeX code 3 times?, can we get around this issue by parameterizing this with number of nodes?

Reply:
In CloudFormation template we cannot specify the number of EC2 instance dynamically, due to which we have to hard code it.